tutorials/tts/NeMo_TTS_Primer.ipynb

```diff
@@ -0,0 +1,1736 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "h19_oQ1GSMe8",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "**HiFi-GAN**\n",
+    "\n",
+    "The architecture of [HiFi-GAN](https://arxiv.org/abs/2010.05646) is similar to WaveNet. The main difference is that it is smaller, and not auto-regressive. This allows it to reach a speed of around 0.01 RTF; 10,000x faster than WaveNet.\n",
+    "\n",
+    "The main way that HiFi-GAN achieves this speed with minimal sacrifice to perceived audio quality is through *adversarial training*, treating the vocoder like a **generative adversarial network (GAN)**.\n",
+    "\n",
+    "To do this, we jointly train the vocoder with a set of *discriminators* that try to predict whether the synthesized/reconstructed audio is real or not. The goal being to trick the discriminators so they cannot systematically tell the difference between the synthesized audio and the original audio. Doing this allows us to get realistic audio with a significantly smaller model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8HRHKLlJSOw-",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "**WaveRNN**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "nbM716VLSzQg",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Many people tend to avoid RNNs because of how slowly they train compared to CNNs and Transformers.\n",
+    "\n",
+    "However, they are one of the most used architectures in real-world applications and industry due to their small compute requirements and memory footprint at inference time. This makes them ideal for low-latency and low-power situations like *on-device* TTS.\n",
+    "\n",
+    "[WaveRNN](https://arxiv.org/abs/1802.08435) is a popular vocoder for those who want to do spectrogram inversion on-device.\n",
+    "\n",
+    "The model itself manages to minimize compute requirements in several different ways that are too complex to summarize adequately here. But a few highlights would be:\n",
+    "\n",
+    "1.   Instead of using a 2<sup>16</sup> sized softmax, it uses two 2<sup>8</sup> sized softmaxes to predict the first and last 8-bits of the 16-bit audio sample separately.\n",
+    "2.   Uses a *subscaling* system to have the RNN predict 16 audio samples at a time instead of 1 sample at a time.\n",
+    "3.   Uses *sparse RNN training* during which the smallest model weights in the RNN are periodically set to zero. Inference is then done efficiently using sparse vectors."
+   ]
```

Comment:
Ends a bit abruptly, could you add a small conclusion section here? Perhaps point to another tutorial that may be useful (the model inference one for people to play around with pretrained models?) as well.

Reply:
+1 for example, link to Riva docs, PyTorch lightning/config/hydra NeMo tutorials, and links to text processing docs. Also, a reference to this paper for more details should be helpful https://arxiv.org/abs/2106.15561

Reply:
Added an extra section at the end linking to TTS documentation/tutorials/riva. I think things like pytorch lightning/hydra would be more appropriate for general NeMo documentation.